src/hotspot/share/jfr/metadata/metadata.xsd

```diff
@@ -0,0 +1,125 @@
+              <xs:attribute name="period" type="periodType" use="optional" />
+              <xs:attribute name="cutoff" type="xs:boolean" use="optional" />
+              <xs:attribute name="throttle" type="xs:boolean" use="optional" />
+            </xs:complexType>
+          </xs:element>
+          <xs:element maxOccurs="unbounded" name="Type">
+            <xs:complexType>
+              <xs:sequence>
+                <xs:element maxOccurs="unbounded" name="Field">
+                  <xs:complexType>
+                    <xs:attribute name="type" type="xs:NMTOKEN" use="required" />
+                    <xs:attribute name="struct" type="xs:boolean" use="optional" />
+                    <xs:attribute name="array" type="xs:boolean" use="optional" />
+                    <xs:attribute name="name" type="xs:NMTOKEN" use="required" />
+                    <xs:attribute name="contentType" type="xs:string" use="optional" />
+                    <xs:attribute name="label" type="xs:string" use="required" />
+                    <xs:attribute name="description" type="xs:string" use="optional" />
+                    <xs:attribute name="experimental" type="xs:boolean" use="optional" />
+                    <xs:attribute name="internal" type="xs:boolean" use="optional" />
+                    <xs:attribute name="relation" type="xs:string" use="optional" />
+                  </xs:complexType>
+                </xs:element>
+              </xs:sequence>
+              <xs:attribute name="name" type="xs:NMTOKEN" use="required" />
+              <xs:attribute name="label" type="xs:string" use="optional" />
+              <xs:attribute name="experimental" type="xs:boolean" use="optional" />
+            </xs:complexType>
+          </xs:element>
+          <xs:element name="XmlType">
+            <xs:complexType>
+              <xs:attribute name="name" type="xs:NMTOKEN" use="required" />
+              <xs:attribute name="contentType" type="xs:NMTOKEN" use="optional" />
+              <xs:attribute name="javaType" type="xs:NMTOKEN" use="optional" />
+              <xs:attribute name="unsigned" type="xs:boolean" use="optional" />
+              <xs:attribute name="parameterType" type="xs:string" use="required" />
+              <xs:attribute name="fieldType" type="xs:string" use="required" />
+             </xs:complexType>
+          </xs:element>
+          <xs:element name="XmlContentType">
+            <xs:complexType>
+              <xs:attribute name="name" type="xs:NMTOKEN" use="required" />
+              <xs:attribute name="annotation" type="xs:string" use="required" />
+            </xs:complexType>
+          </xs:element>
+          <xs:element name="Relation">
+            <xs:complexType>
+              <xs:attribute name="name" type="xs:NMTOKEN" use="required" />
+            </xs:complexType>
+          </xs:element>
+        </xs:choice>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+</xs:schema>
```

Comment:
The version from upstream does not have the final newline: please remove it from this change.